scripts/dnd5e-helpers.js

```diff
@@ -893,24 +933,56 @@ Hooks.on("createOwnedItem", (actor, item, sheet, id) => {
 
 
 
-function ReactionDetect_preCreateChatMessage(msg){
+function ReactionDetect_preCreateChatMessage(msg) {
   const itemId = $(msg.content).attr("data-item-id");
 
   /** could not find the item id, must not have been an item */
-  if (itemId == undefined){
+  if (itemId == undefined) {
     return;
   }
 
   const speaker = getProperty(msg, "speaker");
-  if (speaker){
+  if (speaker) {
     /** hand over to reaction apply logic (checks combat state, etc) */
     ReactionApply(speaker.actor, speaker.token, itemId);
   }
-}
+};
 
 Hooks.on("preCreateChatMessage", async (msg, options, userId) => {
- const reactMode = game.settings.get('dnd5e-helpers', 'cbtReactionEnable');
-  if ( reactMode === 1 || reactMode === 3) {
+  const reactMode = game.settings.get('dnd5e-helpers', 'cbtReactionEnable');
+  if (reactMode === 1 || reactMode === 3) {
     ReactionDetect_preCreateChatMessage(msg);
   }
+
+  let rollType = getProperty(msg, "flags.dnd5e.roll.type");
+  let itemRoll = getProperty(msg, "flags.dnd5e.itemId");
+  if (rollType === "death" && (game.settings.get('dnd5e-helpers', 'owDeathSave'))) {
+    if (parseInt(msg.content) < 6) {
+      let actor = game.actors.get(msg.speaker.actor);
+      OpenWounds(actor, "from a failed death saving throw");
+    }
+  }
+
+
+  if (rollType === "attack" && itemRoll !== undefined && (game.settings.get('dnd5e-helpers', 'owCrit'))) {
+    let rollResult = msg.roll.match(/"result":[0-9]{1,2}/);
+    if (rollResult[2] === "20") {
```

Comment:
Since this is a homebrew rule (right? 😅 ). I'm worried that hard-coding a 20 for crit is going to immediately be a problem.